server/statistics_handler.go

```diff
@@ -30,12 +39,12 @@ type StatsHandler struct {
 func (s *Server) newStatsHandler() *StatsHandler {
 	store, ok := s.driver.(*TiDBDriver)
 	if !ok {
-		panic("Illegal driver")
+		logutil.Logger(context.Background()).Error("Illegal driver")
```

Comment:
I think you should let it panic here. if !ok, `store` may be nil, it will panic in the next procession.

BTW, newStatsHandler only called when we bootstrap the tidb-server, so it's safe to panic.